### DISCUSSION TITLE
[Q] diesel_cli for firebird?

### DISCUSSION THREAD

**Discussion by rafael-arreola** [diesel-rs/diesel#3422] [Q&A] [answered]
is it possible to use the firebird adapter that was developed by [fernandobatels](https://github.com/fernandobatels/rsfbclient) to run diesel_cli migrations?

**Comment by weiznich** [comment, marked as answer]
That's currently not possible, as `diesel_cli` misses the required functionality. It should be possible to execute migrations using the `diesel_migrations` crate assuming the firebird adapter implements all required traits.

That written: I'm open to extend `diesel_cli` such that it would be possible for third party backend implementations to provide plugins to add support for their database systems. I envision something similar than third party cargo commands there. That would require developing a plugin interface for such extensions. I'm happy to see suggestions/development in that area.
